src/query/src/parser.rs

```diff
@@ -142,6 +200,83 @@ fn max_system_timestamp() -> SystemTime {
         .unwrap()
 }
 
+/// The implementation of the `Analyze` extension AST node
```

Comment:
@waynexia I was reflecting on the `Analyze` and `Explain` code as it has a lot in common, WDYT if we come up with a `macro` here?
I've seen macros are used across the code but mostly in tests so not sure if this is a good use case.

Reply:
It's ok to do so. The code are much almost same here 👍